CMakeLists.txt

```diff
@@ -255,7 +255,7 @@ target_compile_features(triton-python-backend PRIVATE cxx_std_${TRITON_MIN_CXX_S
 target_compile_options(
   triton-python-backend PRIVATE
   $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
-    -Wall -Wextra -Wno-unused-parameter -Wno-type-limits -Werror>
```

Comment:
Out of my curiosity why we have to remove this flag for `python_backend`?

Out of JIRA notes I see there is deprecation notice, that may cause a failure in a future. 
It must be assigned to core team. 

```
tmp/tritonbuild/python/build/boost/container_hash/hash.hpp:132:33: error: 'template<class _Arg, class _Result> struct std::unary_function' is deprecated [-Werror=deprecated-declarations]
  132 |         struct hash_base : std::unary_function<T, std::size_t> {};
```

@statiraju , can some one check it?